src/index.d.ts

```diff
@@ -0,0 +1,395 @@
+declare module 'react-native-onesignal' {
+    /* O P T I O N   T Y P E   V A L U E S */
+    // 0 = None, 1 = Fatal, 2 = Errors, 3 = Warnings, 4 = Info, 5 = Debug, 6 = Verbose
+    export type LogLevel = 0 | 1 | 2 | 3 | 4 | 5 | 6;
+
+    // 0 = NotDetermined, 1 = Authorized, 2 = Denied, 3 = Provisional, 4 = Ephemeral
+    export type IosPermissionStatus = 0 | 1 | 2 | 3 | 4;
+
+    // 0 = NotificationClicked, 1 = ButtonClicked
+    export type OpenedEventActionType = 0 | 1;
+
+    /* O B S E R V E R  C H A N G E  E V E N T S */
+    export interface ChangeEvent<T> {
+        from : T;
+        to   : T;
+    }
+
+    export interface PermissionChange {
+        status                  ?: IosPermissionStatus;    // ios
+        hasPrompted             ?: boolean;   // ios
+        provisional             ?: boolean;   // ios
+        areNotificationsEnabled ?: boolean;   // android
+    };
+
+    export interface SubscriptionChange {
+        userId                  : string;
+        pushToken               : string;
+        isSubscribed            : boolean;
+        isPushDisabled          : boolean;
+    };
+
+    export interface EmailSubscriptionChange {
+        emailAddress        ?: string;
+        emailUserId         ?: string;
+        isEmailSubscribed   : boolean;
+    };
+
+    /* N O T I F I C A T I O N S */
+    export interface OSNotification {
+        body            : string;
+        sound           : string;
+        title           : string;
+        launchURL       : string;
+        rawPayload      : object | string;
```

Comment:
I would expect this to just be one type always.

Reply:
this would require a change in the bridges as they return different types across iOS & Android

Reply:
we can change this later on in its own PR